protostar/starknet_gateway/abi_resolver.py

```diff
@@ -0,0 +1,51 @@
+from typing import Optional
+
+from starknet_py.proxy.contract_abi_resolver import (
+    ContractAbiResolver,
+    ProxyResolutionError,
+    ProxyConfig,
+)
+from starknet_py.net.client import Client
+from starknet_py.proxy.proxy_check import (
+    OpenZeppelinProxyCheck,
+    ArgentProxyCheck,
+    ProxyCheck,
+)
+
+from protostar.starknet import AbiType, Address
+
+
+class AbiResolver:
+    def __init__(self, client: Client) -> None:
+        self._client = client
+
+    async def resolve(self, address: Address) -> Optional[AbiType]:
```

Comment:
I would remove this class and replaced it with `get_abi` function, it much better represents the purpose. Resolve is internal starknet py name and in our context it is not obvious what this class does.

Reply:
Resolver is a generic name, not only specific to Starknet.py. This class finds a proper contract and then fetches the ABI. You can suggest a better for the class.

Reply:
It shouldn't be a class it should be a function

Reply:
Discussed on slack, leave it as it is